File: server/controllers/pages.js
import * as db from '../utils/connectDB.js';
const { Client } = require('pg');


export const getHomePage = async(req, res) => {
    const GREEN = '&#x1F7E9';
    const YELLOW = '&#x1F7E8';
    const RED = '&#x1F7E7';
    const BLACK = '&#x1F7E6';

    const results = (await db.runQuery(`
        SELECT pl.name, pl.total_spaces, COUNT(ps.id) FILTER(WHERE ps.occupied = false) AS available_spaces
        FROM ParkingLots pl
        LEFT JOIN ParkingSpaces ps ON pl.id = ps.lot_id
        GROUP BY pl.id
        ORDER BY pl.name
    `)).rows.map(row => ({
        name: row.name,
        available_spaces: parseInt(row.available_spaces, 10),
        total_spaces: parseInt(row.total_spaces, 10),
        icon: row.available_spaces / row.total_spaces > 0.50 ? GREEN :
              row.available_spaces / row.total_spaces > 0.25 ? YELLOW :
              row.available_spaces / row.total_spaces > 0 ? RED :
              BLACK,
    }));

    res.status(200).render('home', { results });
};

export const getMapPage = async(req, res) => {
    res.status(200).render('map');
}

export const getQueryPage = async(req, res) => {
    const results = (await db.runQuery(`
        SELECT pl.name, pl.total_spaces, COUNT(ps.id) FILTER(WHERE ps.occupied = false) AS available_spaces
        FROM ParkingLots pl
        LEFT JOIN ParkingSpaces ps ON pl.id = ps.lot_id
        GROUP BY pl.id
        ORDER BY pl.name
    `)).rows.map(row => ({
        name: row.name,
        available_spaces: parseInt(row.available_spaces, 10),
        total_spaces: parseInt(row.total_spaces, 10),
        icon: row.available_spaces / row.total_spaces > 0.50 ? GREEN :
              row.available_spaces / row.total_spaces > 0.25 ? YELLOW :
              row.available_spaces / row.total_spaces > 0 ? RED :
              BLACK,
    }));


    // Format data for Chart.js
    const chartData = {
        labels: data.map(row => row.timestamp),
        datasets: [{
            label: 'Capacity',
            data: data.map(row => row.capacity),
            borderColor: 'rgb(255, 99, 132)',
            backgroundColor: 'rgba(255, 99, 132, 0.5)'
        }]
    };


    res.status(200).render('query', { chartData });
}

File: website/server/routes/pages.py
from flask import Blueprint, render_template
from sqlalchemy import text

from server.extensions import db


bp = Blueprint("pages", __name__)

@bp.route("/")
def home_page():
    rows = [
        ("Main Lot", 100, 77),
        ("Bauccio Commons Lot", 50, 10),
        ("Waldschmidt Lot", 75, 33),
        ("Shiley Lot", 120, 120),
        ("Christie Lot", 60, 0),
        ("Kenna Lot", 30, 0),
        ("Portsmouth Lot", 40, 20),
    ]
    return render_template("home.html", rows=rows)
    # result = db.session.execute(text("""
    #     SELECT pl.name, pl.total_spaces, COUNT(ps.id)
    #     FILTER(WHERE ps.occupied = false) AS available_spaces
    #     FROM ParkingLots pl LEFT JOIN ParkingSpaces ps ON pl.id = ps.lot_id
    #     GROUP BY pl.id ORDER BY pl.name
    # """))
    # rows = result.all()
    #
    # return render_template("home.html", page_title="Home", rows=rows)
